logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetAvatarAssetUseCase.kt

```diff
@@ -35,17 +36,25 @@ interface GetAvatarAssetUseCase {
     suspend operator fun invoke(assetKey: UserAssetId): PublicAssetResult
 }
 
-internal class GetAvatarAssetUseCaseImpl(private val assetDataSource: AssetRepository) : GetAvatarAssetUseCase {
+internal class GetAvatarAssetUseCaseImpl(
+    private val assetDataSource: AssetRepository,
+    private val userRepository: UserRepository
+) : GetAvatarAssetUseCase {
     override suspend fun invoke(assetKey: UserAssetId): PublicAssetResult =
         // TODO(important!!): do local lookup for the profile pic before downloading a new one
         assetDataSource.downloadPublicAsset(assetKey.value, assetKey.domain).fold({
-            PublicAssetResult.Failure(it)
+            if (it.isNotFoundFailure) {
+                userRepository.removeUserBrokenAsset(assetKey)
+                PublicAssetResult.Failure(it, false)
+            } else {
+                PublicAssetResult.Failure(it, true)
+            }
```

Comment:
Do we need some updates in unit tests to cover this logic? If not then it would be beneficial to add some tests for it 😄